Makefile.cflags

```diff
@@ -66,3 +66,20 @@ endif
 ifeq ($(origin ARFLAGS),default)
   ARFLAGS = rcs
 endif
+
+# check if we want to make a build without any optimization
+ifeq (1,$(ALL_DEBUG_BUILD))
+  CFLAGS += -g3 -gdwarf-3
+
+  # we may only disable optimization if it's not prohibited
+  ifneq (1,$(MUST_OPTIMIZE))
+    # check if the compiler supports optimize for debug
+    ifeq ($(shell $(CC) -Og -E - 2>/dev/null >/dev/null </dev/null; echo $$?),0)
```

Comment:
Then this should be fixed in various places in the make scripts ;)
On 15 Apr 2016 08:36, "Cenk Gündoğan" <notifications@github.com> wrote:

> In Makefile.cflags
> <https://github.com/RIOT-OS/RIOT/pull/4413#discussion_r59832156>:
>
> > @@ -66,3 +66,20 @@ endif
> >  ifeq ($(origin ARFLAGS),default)
> >    ARFLAGS = rcs
> >  endif
> > +
> > +# check if we want to make a build without any optimization
> > +ifeq (1,$(ALL_DEBUG_BUILD))
> > +  CFLAGS += -g3 -gdwarf-3
> > +
> > +  # we may only disable optimization if it's not prohibited
> > +  ifneq (1,$(MUST_OPTIMIZE))
> > +    # check if the compiler supports optimize for debug
> > +    ifeq ($(shell $(CC) -Og -E - 2>/dev/null >/dev/null </dev/null; echo $$?),0)
>
> independent from the idea behind, you could use &>/dev/null instead of 2>/dev/null
> >/dev/null
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/RIOT-OS/RIOT/pull/4413/files/68bed2f85c3295346aa0ac426f5035ed51cafa4f#r59832156>
>